### PR TITLE
Cap the size of originals kept in the proxy cache

### PR DESCRIPTION
**Pull request by feruzm**
## Problem

A proxy miss writes two entries to the proxy store: the rendered variant, and the untouched original. The original is pure cache. It exists only so a later request for a different size or format can re-render locally instead of refetching upstream.

That second write is the expensive one. Sampling 49,366 live files from the imh4 proxy store:

| percentile | size |
|---|---|
| p50 | 69 KB |
| p75 | 310 KB |
| p90 | 1044 KB |
| p95 | 1895 KB |
| p99 | 5116 KB |

Files over 1MB are 10.4% of the count and **67.1% of the bytes**.

On imh4 that tail is why the proxy store takes in ~39 GB/h against 820 GiB of capacity, roughly its own size per day. The store cannot retain a day of traffic, so LRU eviction has nothing old enough to drop and the disk climbs regardless.

## Change

Skip caching the original when it is larger than `max_cached_original_size` (default 1MB, `0` disables caching originals entirely). The variant is always cached, so repeat hits on the same URL are unaffected; only a second, *different* variant of a large source image pays one extra upstream fetch.

At the 1MB default this still caches ~90% of originals by count while dropping the bytes that dominate the store.

The four-clause condition was duplicated across both fetch paths, so it is now `shouldCacheOriginal()` with unit tests rather than a copy-paste.

## Notes

- `config/test.toml` pins the cap to the upload limit. `test/test.jpg` is ~5MB and several existing tests re-render it into a second variant with the source server switched off, which only works if the original was cached. The production default is covered by the `shouldCacheOriginal` unit tests instead.
- Archive stores are untouched: uploads and retention still bypass this path entirely.
- Test suite: 244 passing. The 4 failures in `test/utils.ts` and `test/serve.ts` are present on `main` before this branch and are unrelated.

## Summary by CodeRabbit

* **New Features**
  * Added a configurable size limit for caching original source images alongside rendered variants.
  * Originals up to 1 MB are cached by default; setting the limit to `0` disables original caching.
  * Certain image types remain excluded from original caching regardless of size.

* **Documentation**
  * Expanded configuration documentation with cache behavior, size guidance, and examples.

### CODE REVIEW

File: src/proxy.ts [hunk 1]
```diff
@@ -54,6 +54,24 @@ const INVALIDATE_TOKEN = config.has('invalidate_token')
...
+ * ones are skipped rather than allowed to dominate the cache (see
+ * MAX_CACHED_ORIGINAL_SIZE).
+ */
+export function shouldCacheOriginal(
+    bytes: number,
+    opts: {isDefaultImage: boolean, usesUploadStore: boolean, isLegacy: boolean},
+): boolean {
+    if (opts.isDefaultImage || opts.usesUploadStore || opts.isLegacy) { return false }
+    return bytes <= MAX_IMAGE_SIZE && bytes <= MAX_CACHED_ORIGINAL_SIZE
+}
```
[the remark this answers is not in the document]
Comment (feruzm) [reply src/proxy.ts:78]: Valid, fixed in 7453dce. `shouldCacheOriginal` now returns early on `cap <= 0` before any size comparison, so a zero cap rejects every original including a zero-byte body. The cap is an optional third parameter so the branch is reachable from tests without a second config fixture, and `should treat a zero cap as "never cache originals"` covers it.

[the remark this answers is not in the document]
Comment (feruzm) [reply src/proxy.ts:78]: Same issue CodeRabbit raised, fixed in 7453dce with an explicit `if (cap <= 0) { return false }` ahead of the comparison rather than a strict less-than, so the boundary value itself stays cacheable at a non-zero cap. Now unit tested.

File: test/proxy.ts [hunk 2]
```diff
@@ -249,4 +251,23 @@ describe('proxy', function() {
...
+        const cacheable = {isDefaultImage: false, usesUploadStore: false, isLegacy: false}
+
+        it('should cache originals up to the cap', function() {
+            assert.equal(shouldCacheOriginal(1, cacheable), true)
+            assert.equal(shouldCacheOriginal(MAX_CACHED_ORIGINAL_SIZE, cacheable), true)
+        })
+
+        it('should skip originals over the cap', function() {
+            assert.equal(shouldCacheOriginal(MAX_CACHED_ORIGINAL_SIZE + 1, cacheable), false)
+        })
```
[the remark this answers is not in the document]
Comment (feruzm) [reply test/proxy.ts]: Good catch, this was a real defect in the test. Fixed in 7453dce: the test cap drops from 20000000 to 10000000, which stays above the ~5MB `test/test.jpg` fixture the store tests depend on while sitting strictly below `max_image_size`, so an over-cap size no longer trips the upload gate first.

Also added `should keep the test cap below the upload limit`, asserting `MAX_CACHED_ORIGINAL_SIZE < MAX_IMAGE_SIZE`, so a future config change cannot silently return the suite to proving nothing.

Verified by mutation: deleting `&& bytes <= cap` from `shouldCacheOriginal` now fails `should skip originals over the cap but within the upload limit` (5 failing against the 4 pre-existing baseline). Before this change that mutation passed, exactly as you described.
